File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l=0
        r=len(height)-1

        area = 0
        while (l<r):
            new_area = (r-l)*min(height[l],height[r])
            area = max(area, new_area)
            if (height[l] > height[r]):
                r -= 1
            else:
                l += 1
        return area
        

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int left = 0;
        int right = n-1;
        int maxArea = 0;
        
        int curArea = 0;
        while (left < right){
            curArea = min(height[left],height[right]) * (right - left);
            maxArea = max(maxArea, curArea);
            if (height[left] < height[right]){
                left++;
            }else {
                right--;
            }
        }
        return maxArea;
    }
};